newrelic/hooks/framework_django.py

```diff
@@ -152,38 +152,21 @@ def should_add_browser_timing(response, transaction):
     return True
 
 
-# Response middleware for automatically inserting RUM header and
-# footer into HTML response returned by application
+# Response middleware for automatically inserting RUM header into HTML response returned by application
 
 
 def browser_timing_insertion(response, transaction):
-
-    # No point continuing if header is empty. This can occur if
-    # RUM is not enabled within the UI. It is assumed at this
-    # point that if header is not empty, then footer will not be
-    # empty. We don't want to generate the footer just yet as
-    # want to do that as late as possible so that application
-    # server time in footer is as accurate as possible. In
-    # particular, if the response content is generated on demand
-    # then the flattening of the response could take some time
-    # and we want to track that. We thus generate footer below
-    # at point of insertion.
-
-    header = transaction.browser_timing_header()
```

Comment:
No it's already handled.

We want to keep the actual call to browser timing header as close to insertion as possible because the header includes application server time which is supplied to the browser agent. For that reason we avoid calling it until the very last second, which we do by passing it in as a callable and not an already computed string.

This code from `insert_html_snippet` provides the same functionality we previously had for the header.

```python
    # We are definitely doing to insert something now, so
    # generate the text to be inserted. Bail out if is empty but
    # return the data to indicate we did insert something.

    text = html_to_be_inserted()

    if not text:
        return data
```

Reply:
No it's already handled.

We want to keep the actual call to browser timing header as close to insertion as possible because the header includes application server time which is supplied to the browser agent. For that reason we avoid calling it until the very last second, which we do by passing it in as a callable and not an already computed string.

This code from `insert_html_snippet` provides the same functionality we previously had for the header.

```python
    # We are definitely doing to insert something now, so
    # generate the text to be inserted. Bail out if is empty but
    # return the data to indicate we did insert something.

    text = html_to_be_inserted()

    if not text:
        return data
```